lager/event_loop/qapplication.hpp

```diff
@@ -0,0 +1,48 @@
+//
+// lager - library for functional interactive c++ programs
+// Copyright (C) 2019 Carl Bussey
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#pragma once
+
+#include <QApplication>
+#include <QMetaObject>
+#include <Qt>
+#include <QtConcurrentRun>
+
+#include <functional>
+#include <utility>
+
+namespace lager {
+
+struct with_qapplication_event_loop
+{
+    std::reference_wrapper<QApplication> app;
```

Comment:
I think we could depend in `QCoreApplication` instead, to make it compatible with apps that do not use `QWidgets`, like some QML stuff.

Reply:
Good point. Fixed.